### PR TITLE
[3.3.5] More refactors from the pooling rewrite

### PR DESCRIPTION
**Pull request by Treeston**
Ref #23603. This PR is a no-op in terms of behavior, modulo log output.

* Split `SpawnMetadata` off from `SpawnData`
  * This necessitates splitting the `WorldLocation spawnPoint` into `uint32 mapId` (on `SpawnMetadata`) and `Position spawnPoint` (on `SpawnData`)
  * Associated adjustments make up for a majority of this PR's diff
* `ObjectGridLoader` currently instantiates every creature/gobj, checks its typeid, then deletes it again after checking for respawns. That's really really unnecessary, so it's gone.

### CODE REVIEW

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -1326,9 +1326,15 @@ void Creature::SaveToDB(uint32 mapid, uint8 spawnMask, uint32 phaseMask)
     data.displayid = displayId;
     data.equipmentId = GetCurrentEquipmentId();
     if (!GetTransport())
-        data.spawnPoint.WorldRelocate(this);
+    {
+        data.mapId = GetMapId();
+        data.spawnPoint.Relocate(this);
+    }
     else
-        data.spawnPoint.WorldRelocate(mapid, GetTransOffsetX(), GetTransOffsetY(), GetTransOffsetZ(), GetTransOffsetO());
```
Comment (stanislavkozlovski) [inline src/server/game/Entities/Creature/Creature.cpp]: What's the reason for omitting calls to `spawnPoint#WorldRelocate`? Seems like a decent abstraction

**Comment by rblanca** [comment]
any news about this?...until today pools are broken...objects stop appearing.

**Comment by jackpoz** [comment]
I merged latest 3.3.5 into this PR, fixing the merge conflicts

**Comment by jackpoz** [comment]
@Treeston I'll merge this PR during this weekend

**Comment by Treeston** [comment]
On your responsibility, sure. It was merge ready when I left it, but no idea what changes happened since, I haven't exactly been keeping up with the core.

**Pull request merged**